src/use-swr.ts

```diff
@@ -394,7 +394,7 @@ function useSWR<Data = any, Error = any>(
     const softRevalidate = () => revalidate({ dedupe: true })
 
     // trigger a revalidation
-    if (!config.initialData) {
+    if (!config.initialData && config.revalidateInitialData) {
```

Comment:
Shouldn't this be or?
```suggestion
    if (!config.initialData || config.revalidateInitialData) {
```
If there is not initial data or want to revalidate the initial data, then trigger a revalidation